zombies.js

```diff
@@ -79,136 +100,224 @@
+   *
+   * @name discardItem
+   * @param {Item/Weapon/Food} item   The item to discard.
+   * @return {boolean} true/false     Whether player was able to remove item from pack.
+   */
+
+
+  discardItem(item) {
+    if (this._pack.indexOf(item) === -1) {
+      console.log(item.name + ' not in inventory.');
+      return false
+    } else {
+      this._pack.splice(this._pack.indexOf(item), 1);
+      console.log(this.name + '\'s ' + item.name + ' was discarded.');
+      return true;
+    }
+  }
+
+  /**
+   * Player Class Method => equip(itemToEquip)
+   * -----------------------------
+   * Player equips a weapon item.
+   *
+   * Player can only equip Weapon instances.
+   * Player can only equip weapon items from their pack.
+   *
+   * If the player already has a weapon equipped (the equipped property
+   *   is set to an Item), find the itemToEquip in the pack and replace
+   *   it with the currently equipped item.  Then set the equipped property
+   *   to the itemToEquip.
+   * However, if the player doesn't already have a weapon equipped, simply
+   *   equip that item and remove it from the pack.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name equip
+   * @param {Weapon} itemToEquip  The weapon item to equip.
+   */
+
+  equip(itemToEquip) {
+    let indexOfWeapon = this.getPack().indexOf(itemToEquip);            //find index of weapon to equip in bag
+    if ((itemToEquip instanceof Weapon) && indexOfWeapon !== -1) {      //if item is a weapon and is in bag
+      if (this.equipped === false) {                                     //if currently unequipped
+        this.equipped = itemToEquip;                                    //item is equipped from bag
+        this.discardItem(itemToEquip);                                  //item is taken out from bag
+      } else {
+        this.getPack().splice(indexOfWeapon, 1, this.equipped);         //swap currently equipped weapon with weapon from bag
+        this.equipped = itemToEquip;
+      }
+    }
+  }
+
+  /**
+   * Player Class Method => eat(itemToEat)
+   * -----------------------------
+   * Player eats a food item, restoring their health.
+   *
+   * Player can only eat Food instances.
+   * Player can only eat food items from their pack.
+   *
+   * Remove itemToEat from the pack.
+   * Increase the player's health by the food's energy amount, but do not
+   *   exceed the player's max health.  If exceeded, simply set player's health
+   *   to max health instead.
+   * To access the player's max health, be sure to use Player's getMaxHealth method.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name eat
+   * @param {Food} itemToEat  The food item to eat.
+   */
+
+  eat(itemToEat) {
+    let indexOfFood = this.getPack().indexOf(itemToEat);
+    if ((itemToEat instanceof Food) && indexOfFood !== -1) {
+      if (this.health < this.getMaxHealth()) {
```

Comment:
This if statement is not necessary because we will always consume the food, even at max health.